classes/Product.php

```diff
@@ -3276,11 +3276,16 @@ public static function getPricesDrop(
         $ids_product = Product::_getProductIdByDate((!$beginning ? $current_date : $beginning), (!$ending ? $current_date : $ending), $context);
 
         $tab_id_product = [];
+        
         foreach ($ids_product as $product) {
             if (is_array($product)) {
-                $tab_id_product[] = (int) $product['id_product'];
+                if(!in_array($product['id_product'], $tab_id_product)){
+                    $tab_id_product[] = (int) $product['id_product'];
+                }
             } else {
-                $tab_id_product[] = (int) $product;
+                if(!in_array((int) $product, $tab_id_product)){
+                    $tab_id_product[] = (int) $product;
+                }
```

Comment:
```suggestion
                $productId = (int) $product;
                if(!in_array($productId , $tab_id_product)){
                    $tab_id_product[] = $productId;
                }
```

Do it once for better performance

Reply:
Ok!